front/src/applications/operationalStudies/components/Scenario/Timetable.tsx

```diff
@@ -193,6 +198,61 @@ export default function Timetable({
 
   const timeTableHasInvalidTrain = (trains: ScheduledTrain[]) =>
     trains.some((train) => train.invalid_reasons && train.invalid_reasons.length > 0);
+  const toggleTrainSelection = (id: number) => {
+    const currentSelectedTrainIds = [...selectedTrainIds];
+    const index = currentSelectedTrainIds.indexOf(id); // Find the index of the ID in the array
+
+    if (index === -1) {
+      currentSelectedTrainIds.push(id);
+    } else {
+      currentSelectedTrainIds.splice(index, 1);
+    }
+
+    setSelectedTrainIds(currentSelectedTrainIds);
+  };
+
+  const selectAllTrains = () => {
+    if (
+      trainsList &&
+      trainsList.filter((train: ScheduledTrain) => !train.isFiltered).length ===
+        selectedTrainIds.length
+    ) {
+      setSelectedTrainIds([]);
+    } else {
+      const trainIds: number[] = [];
+      trainsList?.map((train) => trainIds.push(train.id));
+      setSelectedTrainIds(trainIds);
+    }
+  };
+
+  const handleTrainsDelete = async () => {
+    try {
+      const trainsCount = selectedTrainIds.length;
+      await deleteTrainSchedules({ body: { ids: selectedTrainIds } });
```

Comment:
This try/catch doesn't work because we use now RTK (and RTK doesn't throw error). You should add an unwrap() to throw the error I think